### PR TITLE
Listar pr

### PR DESCRIPTION
**Pull request by miguelramirezv**
Pueden revisarlo mañana martes también

### CODE REVIEW

File: app/javascript/components/pr-feed.vue [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  <ul
+    class="card-pr__list"
+  >
+    <li
+      v-for="pullRequest in pullRequests"
+      :key="pullRequest.id"
+    >
+      <div class="card-pr__orientation">
+        <div class="card-pr">
+          <a class="card-pr__name" v-bind:href="pullRequest.html_url">
```
Comment (monkeyci) [inline app/javascript/components/pr-feed.vue:11]: Attribute "bind" should be on a new line         vue/max-attributes-per-line<br>Unexpected 'v-bind' before ':'                   vue/v-bind-style

File: app/javascript/components/pr-feed.vue [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<template>
```
Comment (monkeyci) [inline app/javascript/components/pr-feed.vue:1]: Definition for rule 'vue/max-len' was not found  vue/max-len

File: app/controllers/pull_requests_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    @pull_requests = PullRequest.by_organizations([organization.id]).limit(20)
+    #binding.pry
+  end
+
+  private
+
+  def organization
+    @organization ||= Organization.find_by(login: organization_name)
+  end
+
```
Comment (monkeyci) [inline app/controllers/pull_requests_controller.rb:12]: Trailing whitespace detected.

File: app/controllers/pull_requests_controller.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class PullRequestsController < ApplicationController
+  def index
+    @pull_requests = PullRequest.by_organizations([organization.id]).limit(20)
+    #binding.pry
```
Comment (monkeyci) [inline app/controllers/pull_requests_controller.rb:4]: Missing space after #.

File: app/assets/stylesheets/app/card.scss [hunk 5]
```diff
@@ -159,4 +160,86 @@
...
+    }
+
+    &__orientation {
+      display: flex;
+      flex-direction: row;
+      object-fit: cover;
+    }
+
+    &__list {
+      list-style-type:none;
```
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:240]: Expected single space after ":"                                declaration-colon-space-after

File: app/assets/stylesheets/app/card.scss [hunk 6]
```diff
@@ -159,4 +160,86 @@
...
+      display: block;
+      height: 30px;
+      width: 30px;
+      line-height: 30px;
+
+      -moz-border-radius: 15px; /* or 50% */
+      border-radius: 15px; /* or 50% */
+
+      background-color: $primary-color;
+      color: white;
```
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:206]: Unexpected named color "white"                                 color-named

File: app/assets/stylesheets/app/card.scss [hunk 7]
```diff
@@ -159,4 +160,86 @@
...
+      justify-content: flex-end;
+    }
+
+    &__circle {
+      display: block;
+      height: 30px;
+      width: 30px;
+      line-height: 30px;
+
+      -moz-border-radius: 15px; /* or 50% */
```
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:202]: Unexpected vendor-prefix "-moz-border-radius"                  property-no-vendor-prefix

File: app/assets/stylesheets/app/card.scss [hunk 8]
```diff
@@ -159,4 +160,86 @@
...
+    &__project {
+      margin-top: 15px;
+      flex: 2;
+    }
+
+    &__card {
+      border-radius: 10px;
+      margin-top: 20px;
+      margin-bottom: 1px;
+      margin-right: 0px;
```
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:188]: Unexpected unit                                                length-zero-no-unit

File: app/assets/stylesheets/app/card.scss [hunk 9]
```diff
@@ -159,4 +160,86 @@
...
     }
   }
+  &-pr {
+    @extend .card;
+    border-radius: 10px;
+    margin-top: 20px;
+    margin-bottom: 1px;
+    margin-right: 5px;
+    margin-left: 0px;
+    display: flex;
```
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:170]: Expected newline after ";" in a multi-line declaration block   declaration-block-semicolon-newline-after<br>Unexpected whitespace at end of line                           no-eol-whitespace

File: app/assets/stylesheets/app/card.scss [hunk 10]
```diff
@@ -159,4 +160,86 @@
       }
     }
   }
+  &-pr {
+    @extend .card;
+    border-radius: 10px;
+    margin-top: 20px;
+    margin-bottom: 1px;
+    margin-right: 5px;
+    margin-left: 0px;
```
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:169]: Unexpected unit                                                length-zero-no-unit

File: app/controllers/pull_requests_controller.rb (hunk 4 above)
Comment (monkeyci) [inline app/controllers/pull_requests_controller.rb:4]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Layout/LeadingCommentSpace: Missing space after #.

File: app/controllers/pull_requests_controller.rb (hunk 3 above)
Comment (monkeyci) [inline app/controllers/pull_requests_controller.rb:12]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Layout/TrailingWhitespace: Trailing whitespace detected.

File: spec/controllers/pull_requests_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    let(:organization1) { create(:organization, login: 'my_organization') }
+    let(:organization2) { create(:organization, login: 'other_organization') }
+    let(:repo1) { create(:repository, organization: organization1) }
+    let(:repo2) { create(:repository, organization: organization2) }
+    let!(:pr1) { create(:pull_request, repository: repo1) }
+    let!(:pr2) { create(:pull_request, repository: repo2) }
+
+    before { get :index, params: { organization_name: organization1.login } }
+
+    it { expect(response).to have_http_status(:success) }
```
Comment (monkeyci) [inline spec/controllers/pull_requests_controller_spec.rb:14]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
RSpec/EmptyLineAfterExample: Add an empty line after it.

File: spec/controllers/pull_requests_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    it do
+      binding.pry
```
Comment (monkeyci) [inline spec/controllers/pull_requests_controller_spec.rb:16]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Lint/Debugger: Remove debugger entry point binding.pry.

File: app/javascript/components/pr-feed.vue (hunk 1 above)
Comment (monkeyci) [inline app/javascript/components/pr-feed.vue:11]: **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
'v-bind:href' should be on a new line  vue/max-attributes-per-line

Comment (monkeyci) [inline app/javascript/components/pr-feed.vue:11]: **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Unexpected 'v-bind' before ':'         vue/v-bind-style

File: app/assets/stylesheets/app/card.scss [hunk 13]
```diff
@@ -159,4 +160,86 @@
       }
     }
   }
+  &-pr {
```
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:163]: **[stylelint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Expected empty line before rule                                rule-empty-line-before

File: app/assets/stylesheets/app/card.scss (hunk 10 above)
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:169]: **[stylelint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Unexpected unit                                                length-zero-no-unit

File: app/assets/stylesheets/app/card.scss (hunk 9 above)
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:170]: **[stylelint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Expected newline after ";" in a multi-line declaration block   declaration-block-semicolon-newline-after

Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:170]: **[stylelint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Unexpected whitespace at end of line                           no-eol-whitespace

File: app/assets/stylesheets/app/card.scss (hunk 8 above)
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:188]: **[stylelint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Unexpected unit                                                length-zero-no-unit

File: app/assets/stylesheets/app/card.scss (hunk 7 above)
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:202]: **[stylelint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Unexpected vendor-prefix "-moz-border-radius"                  property-no-vendor-prefix

File: app/assets/stylesheets/app/card.scss (hunk 6 above)
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:206]: **[stylelint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Unexpected named color "white"                                 color-named

File: app/assets/stylesheets/app/card.scss (hunk 5 above)
Comment (monkeyci) [inline app/assets/stylesheets/app/card.scss:240]: **[stylelint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Expected single space after ":"                                declaration-colon-space-after

File: app/views/pull_requests/index.html.erb [hunk 14]
```diff
@@ -0,0 +1 @@
+<%= @pull_requests.last(10).map(&:title) %>
```
Comment (agomezcampero) [inline app/views/pull_requests/index.html.erb]: Está bien solamente tomar los 10 últimos? Se que cuando lo vimos juntos hicimos esto para no mostrar todos, pero quiero validar que el pitch pedía esto también.

File: app/views/pull_requests/index.html.erb [hunk 15]
```diff
@@ -1 +1,7 @@
-<%= @pull_requests.last(10).map(&:title) %>
\ No newline at end of file
+<div id="app" class="app">
+  <%= render 'partials/header' %>
+  <div class="container">
+    <pr-feed :pull-requests="<%= ActiveModel::Serializer::CollectionSerializer.new(@pull_requests, each_serializer: PullRequestSerializer).to_json %>" ></pr-feed>
```
Comment (agomezcampero) [inline app/views/pull_requests/index.html.erb:4]: El serializador podría estar en el controlador, de esa forma esta línea sería solamente

```suggestion
    <pr-feed :pull-requests="<%= @serialized_pull_requests.to_json %>" ></pr-feed>
```

File: app/controllers/pull_requests_controller.rb [hunk 16]
```diff
@@ -1,6 +1,7 @@
 class PullRequestsController < ApplicationController
   def index
-    @pull_requests = PullRequest.by_organizations([organization.id])
+    @pull_requests = PullRequest.by_organizations([organization.id]).limit(20)
```
Comment (agomezcampero) [inline app/controllers/pull_requests_controller.rb:3]: acá si va el comentario, está bien el límite de 20? Me imagino que quedaría para después la paginación

File: app/javascript/components/pr-feed.vue (hunk 1 above)
Comment (agomezcampero) [reply app/javascript/components/pr-feed.vue:11]: es redundante poner el `v-bind,` se puede poner solamente `:href="link"`

File: app/javascript/components/pr-feed.vue [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+            {{ pullRequest.title }}
+          </a>
+          <p class="card-pr__project">
+            {{ pullRequest.repository_name }}
+          </p>
+        </div>
+        <div class="card-pr__card">
+          <p class="card-pr__circle">
+            {{ 0 }}
+          </p>
```
Comment (agomezcampero) [inline app/javascript/components/pr-feed.vue:21]: para que es este 0? Podrías ponerlo sin los {{}} si es solo el número

File: app/controllers/pull_requests_controller.rb (hunk 16 above)
Comment (miguelramirezv) [reply app/controllers/pull_requests_controller.rb:3]: sí está bien para este scope, el último scope va a incluir a todos con paginación

File: app/javascript/components/pr-feed.vue (hunk 17 above)
Comment (miguelramirezv) [reply app/javascript/components/pr-feed.vue:21]: sí, es una valor numérico que va a cambiar con la siguiente PR, lo dejé así por la estructura pero ya va a dejar de ser 0 siempre

File: app/views/pull_requests/index.html.erb (hunk 15 above)
Comment (miguelramirezv) [reply app/views/pull_requests/index.html.erb:4]: ok lo veo!

File: app/views/pull_requests/index.html.erb (hunk 14 above)
Comment (miguelramirezv) [reply app/views/pull_requests/index.html.erb]: sí, esta está outdated, pero igual para este scope estoy mostrando solo 20

File: app/controllers/pull_requests_controller.rb [hunk 18]
```diff
@@ -0,0 +1,17 @@
+class PullRequestsController < ApplicationController
+  def index
+    @pull_requests = PullRequest.by_organizations([organization.id]).limit(20)
+    @serialized_pull_requests = ActiveModel::Serializer::CollectionSerializer.new(
+      @pull_requests, each_serializer: PullRequestSerializer)
```
Comment (monkeyci) [inline app/controllers/pull_requests_controller.rb:5]: Closing method call brace must be on the line after the last argument when opening brace is on a separate line from the first argument.

Comment (monkeyci) [inline app/controllers/pull_requests_controller.rb:5]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Layout/MultilineMethodCallBraceLayout: Closing method call brace must be on the line after the last argument when opening brace is on a separate line from the first argument.

File: app/views/pull_requests/index.html.erb [hunk 19]
```diff
@@ -1 +1,11 @@
...
+<%= @pull_requests.last(10).map(&:title) %>
+=======
+<div id="app" class="app">
+  <%= render 'partials/header' %>
+  <div class="container">
+    <pr-feed :pull-requests="<%= @serialized_pull_requests.to_json %>" ></pr-feed>
+  </div>
+  <%= render "partials/footer" %>
+</div>
+>>>>>>> 3a40e63... fixup! feat(pull-request-controller): add basic index
```
Comment (agomezcampero) [inline app/views/pull_requests/index.html.erb]: Ojo, acá se te quedaron cosas de algún merge que hiciste

Comment (agomezcampero) [reply app/views/pull_requests/index.html.erb]: Vi que los corregiste en el commit siguiente.

**Comment by miguelramirezv** [comment]
[Agregar vista con listado de PR [Scope]](https://trello.com/c/FzHmqznA/218-agregar-vista-con-listado-de-pr-scope)

**Comment by miguelramirezv** [comment]
[Agregar vista con listado de PR [Scope]](https://trello.com/c/FzHmqznA/218-agregar-vista-con-listado-de-pr-scope)

**Comment by miguelramirezv** [comment]
[Se debe poder entrar a ver el listado de PRs de la organización](https://trello.com/c/5e0y1jco/219-se-debe-poder-entrar-a-ver-el-listado-de-prs-de-la-organizaci%C3%B3n)

**Comment by miguelramirezv** [comment]
[Se debe poder acceder a la pull request en GitHub desde el listado de de PRs](https://trello.com/c/hh6B7YFW/220-se-debe-poder-acceder-a-la-pull-request-en-github-desde-el-listado-de-de-prs)

**Pull request merged**